### PR TITLE
fix(ivy): proper i18n postprocessing in case of nested templates

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Prior to this change the postprocess step relied on the order of placeholders combined in one group (e.g. [�#1�|�*1:1�]). The order is not guaranteed in case we have nested templates (since we use BFS to process templates) and some tags are represented using same placeholders. This change performs postprocessing more accurate by keeping track of currently active template and searching for matching placeholder.

Also as a part of this PR I've added a couple tests to make sure ICU-only templates are also handled correctly. That's technically unrelated, but it turned out that no additional changes are required, i.e. just 2 new test cases added.

This PR resolves FW-910.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/render3/i18n.ts [hunk 1]
```diff
@@ -505,43 +513,79 @@ function appendI18nNode(tNode: TNode, parentTNode: TNode, previousTNode: TNode |
...
+      }
+      const [templateId, isCloseTemplateTag, placeholder] = placeholders[idx];
+      if (isCloseTemplateTag) {
+        templateIdsStack.pop();
+      } else if (currentTemplateId !== templateId) {
+        templateIdsStack.push(templateId);
+      }
+      placeholders.splice(idx, 1);
+      return placeholder;
+    });
```
Comment (ocombe) [inline packages/core/src/render3/i18n.ts]: could you add more comments to this block of code? it was a bit hard to review and we might as well make life easier for anyone who comes in the future to change this code

**Review by ocombe — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:230804924:BASE:230804961:1548373604848:c10b6abc)

File: packages/core/src/render3/i18n.ts [hunk 2]
```diff
@@ -505,43 +513,81 @@ function appendI18nNode(tNode: TNode, parentTNode: TNode, previousTNode: TNode |
...
+      const content = phs || tmpl;
+      if (!matches[content]) {
+        const placeholders: PostprocessPlaceholder[] = [];
+        content.split('|').forEach((placeholder: string) => {
+          const match = placeholder.match(PP_TEMPLATE_ID_REGEXP);
+          const templateId = match ? parseInt(match[1], 10) : ROOT_TEMPLATE_ID;
+          const isCloseTemplateTag = PP_CLOSE_TEMPLATE_REGEXP.test(placeholder);
+          placeholders.push([templateId, isCloseTemplateTag, placeholder]);
+        });
+        matches[content] = placeholders;
```
Comment (kara) [inline packages/core/src/render3/i18n.ts]: Nit: I think this block can be broken out into its own function for readability
```suggestion
        matches[content] = getPlaceholders(...);
```

**Review by kara — APPROVED** [review]

**Review by alxhub — APPROVED** [review]

File: packages/core/src/render3/i18n.ts (hunk 2 above)
Comment (AndrewKushnir) [reply packages/core/src/render3/i18n.ts]: Thanks for the review @kara. I think we'll need to refactor this code later (as a part of i18n service work) and move more logic to the compiler, so I'll keep this as is for now.

**Pull request closed without merging**